tools/cmake/set_linker_script.cmake

```
@@ -35,10 +35,11 @@ function(mbed_set_linker_script input_target raw_linker_script_path)
         set(LinkerScriptTarget ${input_target}LinkerScript)
         add_custom_target(${LinkerScriptTarget} DEPENDS ${LINKER_SCRIPT_PATH} VERBATIM)
         add_dependencies(${input_target} ${LinkerScriptTarget})
+        # We hardcore mmemory file to "application"
         target_link_options(${input_target}
             INTERFACE
             "-T" "${LINKER_SCRIPT_PATH}"
-            "-Wl,-Map=${CMAKE_BINARY_DIR}/${APP_TARGET}${CMAKE_EXECUTABLE_SUFFIX}.map"
+            "-Wl,-Map=${CMAKE_BINARY_DIR}/application${CMAKE_EXECUTABLE_SUFFIX}.map"
```

Removing the `APP_TARGET` dependency is great as it was the main roadblock to having multiple CMake targets.

But the hardcode won't help much.

Would using `CMAKE_CURRENT_BINARY_DIR` work? Something like:

```suggestion
            "-Wl,-Map=${CMAKE_CURRENT_BINARY_DIR}/application${CMAKE_EXECUTABLE_SUFFIX}.map"
```

---

If this is causing a trouble, can you create an issue? we use binary dir in multiple places so should be fixed for all.